argilla/src/argilla/datasets/_export/_disk.py

```diff
@@ -90,28 +90,39 @@ def from_disk(
 
         # Get the relevant workspace_id of the incoming dataset
         if isinstance(workspace, str):
-            workspace_id = client.workspaces(workspace).id
-        elif isinstance(workspace, Workspace):
-            workspace_id = workspace.id
+            workspace = client.workspaces(workspace)
+            if not workspace:
+                raise ValueError(f"Workspace {workspace} not found.")
```

Comment:
Should we use the base ArgillaError instead?
```suggestion
                raise ArgillaError(f"Workspace {workspace} not found.")
```